### PR TITLE
Add Config Token and Auth Token

### PR DESCRIPTION
**Pull request by michaelwehrley**
* Create ```ConfigurationToken``` and associate
  instance of with ```Participant```.
* Add callback so token is set on creation.
* Add validation of presence and uniqueness to tokens.
* Allow an instance call to re-generate the token.
* Add model specs for participant and tokens.
* Add `put` to allow for updating token.
* Add controller specs for route and flash messages.
* Include feature test for full stack.
* When a participants sends their request for auth,
  a json response is provided based on the correct
  token and if the token has not expired.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -74,6 +74,15 @@
 
   add_index "configuration_events", ["participant_id"], name: "index_configuration_events_on_participant_id", using: :btree
 
+  create_table "configuration_tokens", force: true do |t|
+    t.datetime "expires_at"
+    t.string   "token",                default: "", null: false
+    t.integer  "participant_id",                    null: false
+    t.datetime "created_at"
+    t.datetime "updated_at"
+    t.string   "authentication_token", default: "", null: false
```
Comment (michaelwehrley) [inline db/schema.rb]: @ericcf I am not sure where to place the authorization token.

File: app/controllers/configuration_tokens_controller.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      flash[:notice] = "Token successfully reset"
+    else
+      flash[:alert] = "Problem reseting token: "\
+                      "#{ @token.errors.full_messages.to_sentence }"
+    end
+    redirect_to root_path
+  end
+
+  def show
+    if @auth_token && @auth_token.valid?
```
Comment (michaelwehrley) [inline app/controllers/configuration_tokens_controller.rb]: @ericcf I am not sure if this is what we are looking for.

File: db/migrate/20150319220116_create_configuration_tokens.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateConfigurationTokens < ActiveRecord::Migration
```
Comment (ericcf) [inline db/migrate/20150319220116_create_configuration_tokens.rb]: @michaelwehrley should this migration add a uniqueness constraint to make it consistent with the model?

File: app/views/dashboards/show.html.erb [hunk 4]
```diff
@@ -20,6 +20,17 @@
...
+              <p>Config Token: <%= participant.token %></p>
+              <% if participant.configuration_token %>
+                <%= link_to "Delete Config Token", participant_configuration_token_path(participant), method: :delete, data: { confirm: 'Are you sure?' } %>
+              <% else %>
+                <%= link_to "Create Config Token", participant_configuration_token_path(participant), method: :post %>
+              <% end %>
+              <% if participant.authentication_token %>
+                <%= link_to "Delete Auth Token", participant_authentication_token_path(participant), method: :delete, data: { confirm: 'Are you sure?' } %>
+              <% else %>
+                <%= link_to "Create Auth Token", participant_authentication_token_path(participant), method: :post %>
```
Comment (ericcf) [inline app/views/dashboards/show.html.erb]: @michaelwehrley no need for manual auth token creation.

File: spec/controllers/authentication_tokens_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,115 @@
+require "rails_helper"
+
+RSpec.describe AuthenticationTokensController, type: :controller do
+  fixtures :all
```
Comment (ericcf) [inline spec/controllers/authentication_tokens_controller_spec.rb]: @michaelwehrley where are these fixtures being used?

Comment (ericcf) [inline spec/controllers/authentication_tokens_controller_spec.rb]: Ah, I see you're signing in a User.

File: db/migrate/20150319220116_create_configuration_tokens.rb (hunk 3 above)
Comment (michaelwehrley) [inline db/migrate/20150319220116_create_configuration_tokens.rb]: yep!

File: app/views/dashboards/show.html.erb (hunk 4 above)
Comment (michaelwehrley) [inline app/views/dashboards/show.html.erb]: Gotcha.

**Comment by michaelwehrley** [comment]
@ericcf I made your recommended updates.

**Comment by michaelwehrley** [comment]
Okay, I updated db.

**Comment by michaelwehrley** [comment]
I guess, after a PR has been closed and I force-pushed an update, I can't reopen the PR.  Please reference: https://github.com/cbitstech/medlink_dashboard/pull/7.  Thanks!

**Pull request closed without merging**